.github/workflows/testing.yml

```diff
@@ -33,13 +47,22 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           fail_ci_if_error: true
-  integration:
-    runs-on: ubuntu-latest
 
+  integration:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, macos-11, macos-12 ]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Install os dependencies
-        run: sudo apt update && sudo apt install wget curl
+        run: |
+          if [ "${{ runner.os }}" = "Linux" ]; then
+            sudo apt update && sudo apt install wget curl
+          else
+            brew install wget curl
```

Comment:
same here
```suggestion
            HOMEBREW_NO_AUTO_UPDATE=1 brew install wget curl
```

Reply:
It's better to set this to Environment and avoid code duplication.

Reply:
Good point!